### PR TITLE
Add new ephemeral package for registering and applying PodOptions changes

### LINKED ISSUE #2856: Add environment variable registration mechanism for ephemeral pods
**Is your feature request related to a problem? Please describe.**
Ephemeral pods launched by Kanister need to have environment variables injected based on the outer OS environment variables (parent pod that launched these ephemeral pods) or based on static environment variables that need to be added. 

The possible set of conditional environment variables needs to be registered and if found in the outer OS then should be injected. The static environment variables, when registered, should always be added to the container environment variables.

**Describe the solution you'd like**
Add an ephemeral package which has a registration scheme for environment variables which can be used outside Kanister to register which environment variables you'd like injected.

**Additional context**
This helps to get the correct FIPS required environment variables into the ephemeral pods to ensure the crypto backend is started in FIPS mode.

### PR DESCRIPTION
**Pull request by kidgilson**
## Change Overview

This PR introduces the ephemeral package meant to handle registering and applying ephemeral pod specific fields.

The first use-case was to introduce FIPS related environment variables that need to be set on all ephemeral pods to ensure the crypto backend (OpenSSL in this case) is started with the FIPS provider enabled and fails otherwise.

The registration mechanism is made to be generic by using an `Applier` interface which is used to register different ways of applying changes to a `*kube.PodOptions` or a `*corev1.Container` .

For now convenience functions related to registering environment variable Applier's are added. The conditional environment variable helper is the OSEnvVar function which looks to see if the environment variable is set on the OS and applies it to the ephemeral pod if so. This usage is the main need for setting the FIPS environment variables since they're already present on the parent pod launching the ephemeral pods.

All places that used the `kube.PodOptions` and `corev1.Container` types have the ephemeral.Options.Apply() function added which adds all registered Appliers (only environment variables at this point) to the pod.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [X] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #2856 

## Test Plan

- [ ] :muscle: Manual
- [X] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Review by bathina2 — APPROVED** [review]
LGTM

**Review by shwetha-97 — APPROVED** [review]
LGTM

File: pkg/ephemeral/ephemeral.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+// FiltererFunc is a function which implements the Filterer interface and can be
+// used to generically filter PodOptions to manipulate using the ApplierList.
+type FiltererFunc[T Constraint] func(*T) bool
+
+func (f FiltererFunc[T]) Filter(options *T) bool {
+	return f(options)
+}
+
+// Filter applies the Applier's if the Filterer criterion is met.
```
Comment (pavannd1) [inline pkg/ephemeral/ephemeral.go:90]:
```suggestion
// Filter applies the Appliers if the Filterer criterion is met.
```

File: pkg/ephemeral/ephemeral.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+			return
+		}
+
+		for _, applier := range appliers {
+			applier.Apply(options)
+		}
+	})
+}
+
+// PodOptionsNameFilter is a Filterer that filters based on the PodOptions.Name.
```
Comment (pavannd1) [inline pkg/ephemeral/ephemeral.go:103]: `PodOptions.Name` is the pod name. Wonder if we should explicitly state that in the comment

File: pkg/ephemeral/envvar.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package ephemeral
+
+import (
+	"os"
+
+	"github.com/kanisterio/kanister/pkg/kube"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// OSEnvVar creates an ApplierSet to set an environment variable if its present
```
Comment (pavannd1) [inline pkg/ephemeral/envvar.go:10]:
```suggestion
// OSEnvVar creates an ApplierSet to set an environment variable if it's present
```

File: pkg/ephemeral/ephemeral.go (hunk 2 above)
Comment (kidgilson) [reply pkg/ephemeral/ephemeral.go]: Sounds good. I'll add more to the comment.

**Review by pavannd1 — APPROVED** [review]

**Pull request merged**